src/test/java/org/elasticsearch/plugins/AbstractPluginsTests.java

```diff
@@ -0,0 +1,548 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.plugins;
+
+import com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.collect.Maps;
+import com.carrotsearch.ant.tasks.junit4.dependencies.com.google.common.collect.Sets;
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import org.apache.tools.zip.ZipEntry;
+import org.apache.tools.zip.ZipOutputStream;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.test.ElasticsearchIntegrationTest;
+import org.elasticsearch.test.ElasticsearchIntegrationTest.ClusterScope;
+import org.elasticsearch.test.ElasticsearchIntegrationTest.Scope;
+import org.junit.Test;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.common.settings.ImmutableSettings.Builder.EMPTY_SETTINGS;
+import static org.elasticsearch.plugins.AbstractPluginsTests.ZippedFileBuilder.createZippedFile;
+import static org.hamcrest.Matchers.equalTo;
+
+@ClusterScope(scope = Scope.SUITE, numDataNodes = 0, numClientNodes = 0, transportClientRatio = 0.0)
+public abstract class AbstractPluginsTests extends ElasticsearchIntegrationTest {
+
+    /**
+     * Method to be implemented by sub classes to test different plugin installations.
+     *
+     * @param fqn          the fully qualified name of the plugin
+     * @param zip          Zip file of the plugin
+     * @param files        List of expected files to be installed
+     * @param env          Environment
+     * @throws Exception
+     */
+    protected abstract void testPlugin(String fqn, Path zip, Collection<String> files, Environment env) throws Exception;
+
+    /**
+     * Test a plugin with the following zipped content (see issue #7152):
+     * <p/>
+     * |
+     * +-- bin/
+     *      +-- tool
+     */
+    @Test
+    public void testPluginWithBinOnly() throws Exception {
+        Path zip = createZippedFile(newTempDirPath(), "plugin_with_bin_only.zip")
```

Comment:
Thanks a lot for this! Definitely better to generate that in a temp dir on the fly instead of having those files as part of the git repo!